### DISCUSSION TITLE
How to pass single value extra-vars to a Play at runtime?

### DISCUSSION THREAD

**Discussion by openstreem** [semaphoreui/semaphore#2393] [General]
Hello,

I've been trying to pass a single value extra-var to my Play. It points to a YAML file:, for example: "@vault.yml".

You can't define it in the Environment as the var must be a JSON key/value.

I've tried configuring [ "--extra-vars", "@vault.yml" ] in the "Args" box. I just receive the error "ERROR! the playbook: [ "--extra-vars","\@vault.yml" ] could not be found".

If I leave the "Args" box empty and add [ "--extra-vars", "@vault.yml" ] to the "Advanced -> CLI Arguments" box, the Play will run normally.

My question:

Is there a way to add my JSON structure to the "Args" box, or, make the "Advanced -> CLI Arguments" box persistent?

I have to execute the Play in this fashion because some of my tasks utilize sudo, and I don't want sudoers to set NOPASSWD for ALL users.

Assistance would be greatly appreciated. This is a show-stopper for me as I'd really like to use Semaphore in lieu of other tools. However, I need to be able to kick off Plays like this automatically and not by inputting a parameter manually at runtime.

Thanks for your thoughts and ideas!

**Comment by fiftin** [comment]
> I've tried configuring [ "--extra-vars", "@vault.yml" ] in the "Args" box. I just receive the error "ERROR! the playbook: [ "--extra-vars","@vault.yml" ] could not be found".
>
> If I leave the "Args" box empty and add [ "--extra-vars", "@vault.yml" ] to the "Advanced -> CLI Arguments" box, the Play will run normally.

it looks like a bug. I will try to reproduce this.

**Reply by fiftin** [reply]
`vault.yml` is a file in the repository?

**Reply by openstreem** [reply]
Correct.

Actually, the file is "vault.enc" (encrypted file) instead of "vault.yml". I just referenced it as a normal YAML file in the example to ease any confusion. It lives in my "group_vars" folder in a private repo.
